conan/tools/microsoft/nmaketoolchain.py

```diff
@@ -15,42 +16,122 @@ def __init__(self, conanfile):
         :param conanfile: ``< ConanFile object >`` The current recipe object. Always use ``self``.
         """
         self._conanfile = conanfile
-        self._environment = None
 
-    @property
-    def cl_flags(self):
-        cppflags = []
+        # Flags
+        self.extra_cflags = []
+        self.extra_cxxflags = []
+        self.extra_ldflags = []
+        self.extra_defines = []
+
+        # Defines
+        self._ndebug = None
         build_type = self._conanfile.settings.get_safe("build_type")
-        if build_type in ['Release', 'RelWithDebInfo', 'MinSizeRel']:
-            cppflags.append("/DNDEBUG")
+        if build_type in ["Release", "RelWithDebInfo", "MinSizeRel"]:
+            self._ndebug = "NDEBUG"
+
+        self._build_type_flags = build_type_flags(self._conanfile.settings)
+        self._build_type_link_flags = build_type_link_flags(self._conanfile.settings)
 
-        bt = build_type_flags(self._conanfile.settings)
-        if bt:
-            cppflags.extend(bt)
+        self._cppstd = cppstd_flag(self._conanfile.settings)
+        self._msvc_runtime_flag = self._get_msvc_runtime_flag()
 
-        cppstd = cppstd_flag(self._conanfile.settings)
-        if cppstd:
-            cppflags.append(cppstd)
-        from conan.tools.microsoft import msvc_runtime_flag
+    def _get_msvc_runtime_flag(self):
         flag = msvc_runtime_flag(self._conanfile)
         if flag:
-            cppflags.append("-{}".format(flag))
-        return " ".join(cppflags).replace("-", "/")
+            flag = f"/{flag}"
+        return flag
+
+    def _sanitized_options(self, options):
+        return [f"{opt[0].replace('-', '/')}{opt[1:]}" for opt in options if len(opt) > 1]
+
+    def _sanitized_defines(self, defines):
+        sanitized_defines = []
+        for define in defines:
+            if "=" in define:
+                # CL env-var can't accept '=' sign in /D option, it can be replaced by '#' sign:
+                # https://learn.microsoft.com/en-us/cpp/build/reference/cl-environment-variables
+                macro, value = define.split("=", 1)
+                if not value:
+                    sanitized_defines.append(f"/D{macro}")
```

Comment:
I've found an issue here while consuming `proj` recipe (it defines `PROJ_DLL=` if static) in a NMake build, `FOO=` definition must be translated to `FOO#` actually.